Add cache item if it doesnt exist yet

roborourke opened the pull request:
In some cases `$batcache->cache` is not an array as it doesnt exist yet but it can subsequently be compared as an array throwing a warning. This ensures if the current page isnt in the cache yet it gets added and prevents the warning.

roborourke committed:
Add cache item if it doesnt exist yet

In some cases `$batcache->cache` is not an array as it doesnt exist yet but it can subsequently be compared as an array throwing a warning. This ensures if the current page isnt in the cache yet it gets added and prevents the warning.

rmccue commented on advanced-cache.php:
```diff
@@ -566,20 +566,21 @@ function add_debug_html_to_output( $debug_html ) {
-		}
+} else if ( ! is_array( $batcache->cache ) ) {
+	// No batcache item found, create it.
+	$batcache->do = true;
```
I think this needs to do the cache add and cache incr too?

roborourke replied:
Are they not just added at the end of page generation? I don't really know the inner workings of this too intimately.

rmccue replied:
Not certain, just it's behaviour that was there before, so we should replicate imo.

rmccue replied:
Oh, I am wrong! It'll be cleared by the `wp_cache_delete` that was there before.

I think all we need to do then is `$batcache->requests = 1`, not the cache entries

roborourke replied:
Fixed!

roborourke committed:
Create the cache entry and incrememnt

roborourke committed:
Set requests value to 1

ntwb commented:
FYI: Automattic have an alternative solution for this in https://github.com/Automattic/batcache/pull/96

Commit:
Fix array offset on value type bool

Fix implemented from Steven's suggestion. Original PR and Link:  Fixed array offset access on value of type bool with PHP 7.4 #96 : https://github.com/Automattic/batcache/pull/96

Commit:
add is_cached variable

From the implementation of this fix https://github.com/Automattic/batcache/pull/96 we needed to add the $is_checked variable.

Commit:
Remove extra space

robindevitt commented:
Fix implemented from @ntwb's [suggestion](https://github.com/humanmade/batcache/pull/16#issuecomment-898066220). Original PR and Link:  Fixed array offset access on value of type bool with PHP 7.4

This fix supercedes https://github.com/humanmade/batcache/pull/16
